app/src/main/java/org/asteroidos/sync/services/SynchronizationService.java

```diff
@@ -306,8 +313,10 @@ else if (mState == STATUS_CONNECTED)
 
     @Override
     public void onDestroy() {
-        if(mDevice != null)
-            mDevice.disconnect();
+        if (mDevice != null)
```

Comment:
no need to check that, the result will be the same if you unconditionally mDevice = null; ;)